src/braket/analog_hamiltonian_simulator/rydberg/rydberg_simulator.py

```diff
@@ -0,0 +1,205 @@
+        self.rydberg_interaction_coef = validate_rydberg_interaction_coef(
+            rydberg_interaction_coef
+        ) / ((SPACE_UNIT**6) / TIME_UNIT)
+        self.blockade_radius = validate_blockade_radius(blockade_radius) / SPACE_UNIT
+
+        # Convert the units of the program from SI unit to microsecond and micrometer etc.
+        program = convert_unit(program)
+
+        # Get the duration of the program. Note that there could be 0 or 1 driving field
+        # (same for the shifting field) for a given program. Hence we need to checkout
+        # the duration for both fields
+        self.duration = 0
+        if len(program.hamiltonian.drivingFields) == 1:
+            self.duration = float(
+                program.hamiltonian.drivingFields[0].amplitude.time_series.times[-1]
+            )
+        elif len(program.hamiltonian.shiftingFields) == 1:
+            self.duration = float(
+                program.hamiltonian.shiftingFields[0].magnitude.time_series.times[-1]
+            )
+
+        if self.duration == 0:
+            self.simulation_times = [0]
+        else:
+            self.simulation_times = np.linspace(0, self.duration, steps)
+
+        # Get valid configurations that comply with the blockade approximation
+        self.atomArray = program.setup.ahs_register
+        self.configurations = get_blockade_configurations(self.atomArray, self.blockade_radius)
+
+        # Run the solver
+        # We shall adaptively change between numpy solver (RK6 method) and scipy solver
+        if len(self.configurations) <= 1000:
+            states = rk_run(
+                program,
+                self.configurations,
+                self.simulation_times,
+                self.rydberg_interaction_coef,
+            )
+        else:
+            states = scipy_integrate_ode_run(
+                program,
+                self.configurations,
+                self.simulation_times,
+                self.rydberg_interaction_coef,
+                progress_bar=progress_bar,
+                atol=atol,
+                rtol=rtol,
+                solver_method=solver_method,
+                order=order,
+                nsteps=nsteps,
+                first_step=first_step,
+                max_step=max_step,
+                min_step=min_step,
+            )
+
+        # Convert the result type
+        if shots == 0:
+            raise NotImplementedError("Shot = 0 is not currently implemented")
+        else:
+            dist = sample_state(states[-1], shots)
+            return convert_result(
+                dist,
+                self.atomArray.filling,
+                self.configurations,
+                self.mock_taskMetadata(shots),
+            )
+
+    def mock_taskMetadata(self, shots: int) -> TaskMetadata:
+        return TaskMetadata(
+            id="rydberg",
+            shots=shots,
+            deviceId="rydbergLocalSimulator",
+        )
```

Comment:
Can this be private? Also ,the name `"mock" should not be appearing in a non-test file.
```suggestion
    def _task_metadata(self, shots: int) -> TaskMetadata:
        return TaskMetadata(
            id="rydberg",
            shots=shots,
            deviceId="rydbergLocalSimulator",
        )
```